### PR TITLE
[Query] Add address resolver for server

### PR DESCRIPTION
**Pull request by gichan-jang**
Add address resolver for server.

Signed-off-by: gichan <gichan2.jang@samsung.com>

Todo:
 common properties are adjusted by .ini and envvar.

Test pieline
Server:
```
gst-launch-1.0 tensor_query_serversrc ! other/tensors,num_tensors=1,dimensions=3:300:300:1,types=uint8,framerate=30/1 ! tensor_query_serversink
```

Client:
```
gst-launch-1.0 v4l2src ! videoconvert ! videoscale !  video/x-raw,width=300,height=300,format=RGB,framerate=30/1 ! tensor_converter ! tensor_query_client ! tensor_decoder mode=direct_video ! videoconvert ! ximagesink
```

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 1]
```diff
@@ -414,7 +431,13 @@ nnstreamer_query_server_init (query_server_handle server_data,
     {
       GSocketAddress *saddr;
       GError *err = NULL;
-      saddr = g_inet_socket_address_new_from_string (host, port);
+
+      sdata->cancellable = g_cancellable_new ();
+      if (!gst_tensor_query_get_saddr (host, port, &saddr, sdata->cancellable)) {
+        nns_loge ("Failed to get socket address");
+        return EADDRNOTAVAIL;
```
Comment (wooksong) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:438]:
```suggestion
        return -EADDRNOTAVAIL;
```

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 2]
```diff
@@ -110,31 +110,29 @@ nnstreamer_query_connection_get_port (query_connection_handle connection)
...
 
+
 /**
- * @brief Create requested socket.
+ * @brief Get socket address
  */
 static gboolean
-gst_tensor_query_socket_new (query_connection_handle conn_h,
-    GSocketAddress ** saddr)
+gst_tensor_query_get_saddr (const char *host, uint32_t port,
```
Comment (wooksong) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:118]: In general, the TCP port field is 16 bits (i.e., int16_t). ```gst_tensor_query_get_saddr ``` also takes ```int16_t port``` for its second argument.

Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Updated to uint16. Thanks.

Comment (wooksong) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: It is not ```uint16_t``` but ```int16_t```.

File: gst/nnstreamer/tensor_query/tensor_query_common.h [hunk 3]
```diff
@@ -107,7 +107,7 @@ nnstreamer_query_connection_get_port (query_connection_handle connection);
  * @return 0 if OK, negative value if error
  */
 extern int
-nnstreamer_query_send (query_connection_handle connection, TensorQueryCommandData *data, uint32_t timeout_ms);
+nnstreamer_query_send (query_connection_handle connection, TensorQueryCommandData *data, uint16_t timeout_ms);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_query/tensor_query_common.h:110]: need rollback this line

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 2 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Well..
`g_inet_socket_address_new` takes `guint16 port`. [link](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/latest/gio/GInetSocketAddress.html#g-inet-socket-address-new)
Do we need negative port number?

File: gst/nnstreamer/tensor_query/tensor_query_common.h (hunk 3 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.h]: Updated.

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 2 above)
Comment (wooksong) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: You are absolutely right! I am confused:(

**Review by wooksong — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 4]
```diff
@@ -110,31 +110,29 @@ nnstreamer_query_connection_get_port (query_connection_handle connection)
...
 
+
 /**
- * @brief Create requested socket.
+ * @brief Get socket address
  */
 static gboolean
-gst_tensor_query_socket_new (query_connection_handle conn_h,
-    GSocketAddress ** saddr)
+gst_tensor_query_get_saddr (const char *host, uint16_t port,
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:118]: Better to use `guint16` instead of `uint16_t`?

Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Updated.

Comment (dongju-chae) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Actually, as `TensorQueryConnection` has declared `port` as `guint16_t`, it's better to unify the type :) `uint16_t port` still exist.

**Review by kimjh12 — APPROVED** [review]
Looks great

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 4 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c:118]: Changed GLib types to standard types. Please check it again.

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 5]
```diff
@@ -41,8 +41,8 @@ typedef struct
       GCancellable *cancellable;
       GAsyncQueue *conn_queue;
     };
-    /* check the size of struct is less */
-    guint8 _dummy[TENSOR_QUERY_SERVER_DATA_LEN];
+    // check the size of struct is less
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:44]: fix code style

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 6]
```diff
@@ -147,14 +145,34 @@ gst_tensor_query_socket_new (query_connection_handle conn_h,
...
+    nns_loge ("Failed to get socket address");
+    return FALSE;
+  }
+
   /* create sending client socket */
   /** @todo Support UDP protocol */
   conn->socket =
       g_socket_new (g_socket_address_get_family (*saddr), G_SOCKET_TYPE_STREAM,
       G_SOCKET_PROTOCOL_TCP, &err);
+  g_object_unref (*saddr);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:175]: plz check this line. (move this to line 177?)

Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Right. It should not be un-referred here. Updated.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**